torch/csrc/utils/tensor_new.cpp

```diff
@@ -531,9 +531,39 @@ Tensor as_tensor(const Type& type, PyObject* args, PyObject* kwargs) {
   ParsedArgs<3> parsed_args;
   auto r = parser.parse(args, kwargs, parsed_args);
   if (r.idx == 0) {
+    at::optional<Device> device_opt = r.deviceOptional(2);
```

Comment:
as mentioned above, I don't think you need any of this.

Reply:
right, I will remove them all together